cowtown2023/src/main/java/com/team2357/frc2023/subsystems/ShooterSubsystem.java

```diff
@@ -24,6 +49,67 @@ public void runShooter(double topPercentOutput, double bottomPercentOutput) {
     public void stopShooter() {
         m_topShooterMotor.set(0.0);
         m_bottomShooterMotor.set(0.0);
+        m_isClosedLoopEnabled = false;
+    }
+
+    public void setClosedLoopEnabled(){
+        m_isClosedLoopEnabled = true;
+    }
+
+     // {degrees, top shooter RPM, bottom shooter RPM}
+    private static final double[][] degreesToRPM = {
+        { 20, 0, 0 },
+        { 15, 0, 0 },
+        { 10, 0, 0 },
+        { 5, 0, 0 },
+        { 0, 0, 0 },
+        { -5.0, 0, 0 },
+        { -10, 0, 0 },
+        { -15, 0, 0 },
+        { -20, 0, 0 }
+    };
+
+    @Override
+    public void periodic(){
+        if(m_isClosedLoopEnabled){
+
+        }
+
     }
 
+    /**
+     * Supposed to set the shooter motors output based on a vision targets angle
+     * @param angle The angle of the target
+     */
+    private void setShooterRPMVision(double angle){
+        int curveSegmentIndex = RobotMath.getCurveSegmentIndex(degreesToRPM, angle);
+        if (curveSegmentIndex == -1) {
+            System.err.println("----- Curve segment index out of bounds -----");
+            return;
+        }
+
+        double[] pointA = degreesToRPM[curveSegmentIndex];
+        double[] pointB = degreesToRPM[curveSegmentIndex + 1];
+
+        double highAngle = pointA[0];
+        double lowAngle = pointB[0];
+        double highBottomRPMs = pointA[1];
+        double lowBottomRPMs = pointB[1];
+        double highTopRPMs = pointA[2];
+        double lowTopRPMs = pointB[2];
+
+        double bottomRpms = RobotMath.lineralyInterpolate(highBottomRPMs, lowBottomRPMs, highAngle, lowAngle, angle);
+
+        double topRpms = RobotMath.lineralyInterpolate(highTopRPMs, lowTopRPMs, highAngle, lowAngle, angle);
```

Comment:
Not your fault but "linearly" is spelt wrong in `lineralyInterpolate`. If you change it here, I'll change it in the robotlib repo.